Lib/test/test_compile.py

```diff
@@ -1130,7 +1130,8 @@ def check_positions_against_ast(self, snippet):
         class SourceOffsetVisitor(ast.NodeVisitor):
             def generic_visit(self, node):
                 super().generic_visit(node)
-                if not isinstance(node, ast.expr) and not isinstance(node, ast.stmt):
+                skip = [ast.expr, ast.stmt, ast.pattern]
+                if all(not isinstance(node, s) for s in skip):
```

Comment:
This is the same, right? I find it much less confusing to follow:

```suggestion
                if not isinstance(node, (ast.expr, ast.stmt, ast.pattern)):
```